force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,9 +197,14 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer generateRandomNumber() {
         // Use Math class to generate number between 50 and 100
+        final Integer MIN = 50;
+        final Integer MAX = 100;
+
+        Integer result = MIN + (Integer)Math.floor(Math.random() * (MAX - MIN + 1));
+        System.debug(result);
```

Comment:
I really like this representation of the problem even with the extra local variables declared because it makes the actual random formula a bit more readable/changeable.